### PR TITLE
Ecf bugs

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/models/tasks/help_request.rb [hunk 1]
```diff
@@ -5,7 +5,10 @@ class HelpRequest < NurseTask
     OVERDUE_AFTER_DAYS = 2
 
     validates :user_id,
-              uniqueness: { scope: [:participant_id, :type, :status] }
+              uniqueness: {
+                scope: [:participant_id, :type],
+                conditions: -> { where(status: NurseTask::STATUSES.active) }
```
Comment (michaelwehrley) [inline app/models/tasks/help_request.rb]: I take it you can only have one "active" request that fits these conditions, but many "resolved"?

Comment (michaelwehrley) [inline app/models/tasks/help_request.rb]: I see the answer in the specs below :smile:

Comment (ericcf) [inline app/models/tasks/help_request.rb]: Yup! I just learned about the `conditions` option today. Pretty handy.

**Comment by michaelwehrley** [comment]
Looks great.  In the future this duplicated code seems like a great place for an `ActiveRecord::Concern`
```ruby
validates :user_id,
  uniqueness: { scope: [:participant_id, :type, :status] }
    uniqueness: {
      scope: [:participant_id, :type],
        conditions: -> { where(status: NurseTask::STATUSES.active) }
    }
  ```

**Pull request merged**